Add support for snappy

alfonsosanchezbeato opened the pull request:
Add support for snappy by checking env vars to know where to store/find data files.

tonyespy commented on plugins/mbpi.c:
```diff
@@ -565,34 +564,39 @@ static gboolean mbpi_parse(const GMarkupParser *parser, gpointer userdata,
+	char *filename;
+
+	snap = getenv("SNAP");
+	filename = g_strdup_printf("%s%s", snap ? snap : "", MBPI_DATABASE);
```
We should discuss where mbpi ends up living as it's used by modemmanager too.

I think it probably should live in it's own snap that shares the file via content sharing.  We also should explore the ability for an end-user to modify this file on a working system ( ie. allow a writable version that would override the default ) vs. forcing the user to report a bug and wait for an official update.

tonyespy commented on plugins/mbpi.c:
```diff
@@ -565,34 +564,39 @@ static gboolean mbpi_parse(const GMarkupParser *parser, gpointer userdata,
+	gboolean ret = FALSE;
+	const char *snap;
+	char *filename;
+
```
Please add a short comment that explains the SNAP env variable.

tonyespy commented on plugins/phonesim.c:
```diff
@@ -1107,12 +1107,19 @@ static void parse_config(void)
 	const char *filename;
+	char *snap_filename = NULL;
+	const char *conf_override = getenv("OFONO_PHONESIM_CONFIG");
+	const char *snap_common = getenv("SNAP_COMMON");
```
Please add a short comment explaining why SNAP_COMMON is used here.

tonyespy commented on src/storage.c:
```diff
@@ -217,3 +217,13 @@ void storage_close(const char *imsi, const char *store, GKeyFile *keyfile,
 
 	g_key_file_free(keyfile);
 }
+
```
Please add a short comment here describing the purpose of this function, and why SNAP_COMMON is used.

tonyespy commented:
Might as well add some extra logic to plugins/u8500.c, as it won't be possible for a user to add the file /etc/imei.

alfonsosanchezbeato commented:
Branch refreshed after addressing comments. Some points:

1. Yes, we agreed on creating a new snappy branch, and also on following the discussion here until the MP is given thumbs up. Let's do that, and create a new MP after that.
2. Take into account that apns-conf.xml is part of the Android container. Before changing ubuntu-apndb we have to add it to ofono. In fact I remember to have discussed about this before, and the conclusion was that we should move the DB to the ofono package. We can discuss this online.
3. Having MBPI as a snap would introduce dependencies, which would go against snappy design.
4. No, MBPI should not be writable by the user, as apns-conf.xml is not either in Touch. If the operator is not present, the user will have to change the provisioning data, which will be saved by ofono anyway, so this needs to be done only once. Being able to modify the DB will not make this less painful for the user. Of course, updating the DB for these operators is great, but we do not need to let the user modify (which might led to corruption) the local DB for this.
